src/Report_Generation/GenerateHTML.java

```diff
@@ -0,0 +1,910 @@
+		String section = "";
+		Set<String> classNames = calc.getMurgePulledValues().getNumberOfClassesInProject();
+		String[][] abstraction = new String[4][6];
+		abstraction[0] = new String[] { "Metric", "Average", "Standerd Deviation", "Higeset Value", "Lowest Value",
+				"Count" };
+		abstraction[1] = generateRow(classNames,calc.getPrimaryMetrics().getAverageNumberOfAncestors(),"Average Number of Ancestors");
+		abstraction[2] = generateRow(classNames,calc.getPrimaryMetrics().getFailSafeDefaults(),"Fail-Safe Defaults");
+		abstraction[3] = generateRow(classNames,calc.getPrimaryMetrics().getReduceAttackSurface(),"Reduce Attack Surface");
+		section += makeTable(abstraction);
+		return section;
+	}
+	public String generateCohesion() {
+		String section = "";
+		Set<String> classNames = calc.getMurgePulledValues().getNumberOfClassesInProject();
+		String[][] cohesion = new String[3][6];
+		cohesion[0] = new String[] { "Metric", "Average", "Standerd Deviation", "Higeset Value", "Lowest Value",
+				"Count" };
+		cohesion[1] = generateRow(classNames,calc.getPrimaryMetrics().getCohesionAmongMethodsInClass(),"Cohesion Among Methods in a Class");
+		cohesion[2] = generateRow(classNames,calc.getPrimaryMetrics().getLackOfCohesionOfMethods(),"Lack of Cohesion of Methods");
+		section += makeTable(cohesion);
+		return section;
+	}
+	public String generateCoupling() {
+		String section = "";
+		Set<String> classNames = calc.getMurgePulledValues().getNumberOfClassesInProject();
+		String[][] coupling = new String[10][6];
+		coupling[0] = new String[] { "Metric", "Average", "Standerd Deviation", "Higeset Value", "Lowest Value",
+				"Count" };
+		HashMap<String,Double> baseClasses = new HashMap<String,Double>();
+		baseClasses.put("project",calc.getPrimaryMetrics().getCountOfBaseClasses());
+		coupling[1] = generateRow(classNames,baseClasses,"Count of Base Classes");
+		coupling[2] = generateRow(classNames,calc.getPrimaryMetrics().getCouplingBetweenObjects(),"Coupling Between Objects");
+		coupling[3] = generateRow(classNames,calc.getPrimaryMetrics().getCouplingCorruptionPropagation(),"Coupling Corruption Propagation");
+		coupling[4] = generateRow(classNames,calc.getPrimaryMetrics().getDepthOfInheritanceTree(),"Depth of Inheritance Tree");
+		coupling[5] = generateRow(classNames,calc.getPrimaryMetrics().getDirectClassCoupling(),"Direct Class Coupling");
+		coupling[6] = generateRow(classNames,calc.getPrimaryMetrics().getFanIn(),"Fan In");
+		coupling[7] = generateRow(classNames,calc.getPrimaryMetrics().getFanOut(),"Fan Out");
+		coupling[8] = generateRow(classNames,calc.getPrimaryMetrics().getHenryKafura(),"Henry Kafura");
+		coupling[9] = generateRow(classNames,calc.getPrimaryMetrics().getNumberOfChildren(),"Number of Children");
+		coupling[9] = generateRow(classNames,calc.getPrimaryMetrics().getNumberOfChildren(),"Responses set for a Class");
+		section += makeTable(coupling);
+		return section;
+	}
+	public String DesignSize() {

```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

Reply:
Thank your for the suggestion, your right about the naming problem, I'll fix that in the next couple of days

Out of Curiosity how did you find this project? Because it's a school project